### DISCUSSION TITLE
A Panic occured in 'internal\core\properties.rs', at line 323, column 13

### DISCUSSION THREAD

**Discussion by tao-a-man** [slint-ui/slint#6075] [Q&A]
I get issue when I update UI of SLint so much times, I use ESP_IDF, Issue very strange, I can't find it on the internet, Plsss help me a idea!!

![Screenshot 2024-01-31 165613](https://github.com/slint-ui/slint/assets/92592956/8ac56f55-5ce8-408e-9e67-eafbba1aaf63)

![Screenshot 2024-01-31 164912](https://github.com/slint-ui/slint/assets/92592956/d2b552e9-305c-402c-a633-65f239cd531c)

**Comment by tao-a-man** [comment]
hii @ogoffart this log in the code `assert_eq!(self.0.get(), local.map_or(null(), |v| v.get_ref() as *const BindingHolder));` line 323, column 13, when i call api get data and update UI with code `auto nodelists = std::make_shared<slint::VectorModel<INode>>(nodes);
                                                    ui->set_NodeIndex(0);
                                                    auto node = nodes[0];
                                                    ui->set_Node(node);
                                                    ui->set_ListNodeEnabel(nodelists);` and code `slint::invoke_from_event_loop([&, ui = slint::ComponentWeakHandle(ui)]
                                                                              {
                                                                                  auto ui_lock = *ui.lock();
                                                                                  ui_lock->fn_HandleBlockScreen(false);
                                                                                  // end ui
                                                                              });` the log happen and i can't call api anymore

**Comment by ogoffart** [comment]
The issue is happening because you are using Slint from different task.
You should dedicate one task to Slint and do all the call to Slint function in that task only.
The only function you can call from another task is [slint::invoke_from_event_loop](https://slint.dev/releases/1.4.0/docs/cpp/api/function_namespaceslint_1a4067c90b05ef4cb20e055117b0263430#namespaceslint_1a4067c90b05ef4cb20e055117b0263430)   (or the blocking version)

**Comment by tao-a-man** [comment]
> The issue is happening because you are using Slint from different task. You should dedicate one task to Slint and do all the call to Slint function in that task only. The only function you can call from another task is [slint::invoke_from_event_loop](https://slint.dev/releases/1.4.0/docs/cpp/api/function_namespaceslint_1a4067c90b05ef4cb20e055117b0263430#namespaceslint_1a4067c90b05ef4cb20e055117b0263430) (or the blocking version)

Thank you for reply but Plss clearly that, I create `ui` in `app_main()` like `// create ui
    static slint::ComponentHandle<AppWindow> ui = AppWindow::create();` and I use `ui ` variable in different tasks are unreasonable?

**Comment by tao-a-man** [comment]
> The issue is happening because you are using Slint from different task. You should dedicate one task to Slint and do all the call to Slint function in that task only. The only function you can call from another task is [slint::invoke_from_event_loop](https://slint.dev/releases/1.4.0/docs/cpp/api/function_namespaceslint_1a4067c90b05ef4cb20e055117b0263430#namespaceslint_1a4067c90b05ef4cb20e055117b0263430) (or the blocking version)

Hii @ogoffart I just use `ui->set_...` in `Task READER` another Task I use `slint::invoke_from_event_loop([&, ui = slint::ComponentWeakHandle(ui)]
                                                              {
                                                                  auto ui_lock = *ui.lock();
                                                                  ui_lock->set_Notification(1);
                                                                  // end event
                                                              });` like u say but issue still happen
